src/include/cache.i

```diff
@@ -458,6 +458,10 @@ __wt_cache_eviction_check(WT_SESSION_IMPL *session, bool busy, bool readonly, bo
     if (didworkp != NULL)
         *didworkp = false;
 
+    /* Eviction causes reconciliation. So don't evict if we can't reconcile */
+    if (F_ISSET(session, WT_SESSION_NO_RECONCILE))
```

Comment:
I am not 100% sure if allowing eviction of clean pages is relevant here. I will have to further look at the code. Earlier we would have successfully evicted clean pages and failed on dirty pages if no-reconciliation flag is set.

Reply:
@sulabhM : I thought about this a bit more after we talked.  I'm currently thinking this isn't a problem. Or, at least, it won't prevent any eviction we were doing before this PR.  Here's my reasoning:

The most common (only?) scenario where we call `wt_cache_eviction_check()` while we're in reconciliation is via `cursor_enter()` on history store cursor operations.  Before this PR, we would *never* evict on history store cursor operations because we would set `WT_SESSION_IGNORE_CACHE_SIZE` when opening a hs cursor.  

So even if we are missing opportunities here to evict clean pages.  We wouldn't have evicted those pages before this change.  

That still leaves a question of whether we really are missing opportunities to evict clean pages here, and if so how we could change the code to allow them.  But that strikes me as an opportunity for further optimization, not as a problem that this change would introduce.  

If that makes sense to you, I can merge this PR and create a new ticket to investigate/optimize clean page eviction here.  Or if I've missed something (quite possible; it's late here :-) I can go ahead with the original plan to ping @agorrod  for an opinion about this.

Reply:
This makes sense to me. We can create a separate ticket for further optimization. I think we can go-ahead with the merge.